pkgs/development/python-modules/mmcv/default.nix

```diff
@@ -0,0 +1,92 @@
+{ lib
+, buildPythonPackage
+, fetchFromGitHub
+, pytest
+, pythonOlder
+, cudaSupport ? false
+, cudatoolkit
+, torchCudaArchList ? "Kepler"
+, torch
+, torchvision
+, torch-bin
+, torchvision-bin
+, opencv4
+, yapf
+, packaging
+, pillow
+, addict
+, ninja
+, which
+, onnx
+, onnxruntime
+, scipy
+, pyturbojpeg
+, tifffile
+, lmdb
+}:
+
+buildPythonPackage rec {
+  pname = "mmcv";
+  version = "1.7.0";
+  format = "setuptools";
+
+  disabled = pythonOlder "3.6";
+
+  src = fetchFromGitHub {
+    owner = "open-mmlab";
+    repo = pname;
+    rev = "refs/tags/v${version}";
+    hash = "sha256-EVu6D6rTeebTKFCMNIbgQpvBS52TKk3vy2ReReJ9VQE=";
+  };
+
+  torch-deps = if cudaSupport then [ torch-bin torchvision-bin ] else [ torch torchvision ];
+  nativeBuildInputs = [ ninja which ]
+    ++ lib.lists.optionals cudaSupport [ cudatoolkit.lib cudatoolkit.out cudatoolkit.cc ] ++ torch-deps;
```

Comment:
it shouldn't be necessarily for torch/torchvision to be in nativeBuildInputs and propagatedBuildInputs.

Reply:
@bcdarwin torch is needed at runtime. not sure there is another way to do it without propagatedBuildInputs